### DISCUSSION TITLE
Enable / Disable Option for all Oqtane Module Forms

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3287] [General] [answered]
Hi,

I would like to enable or disable all module forms (Oqtane built-in forms & custom module forms) based on the configuration. What is the recommended way to do this feature?

**Note:** Enable / Disable radio button will be placed in the Admin Panel.

![image](https://github.com/oqtane/oqtane.framework/assets/42769344/dbf122fa-f229-4c46-a11d-820de44483e6)

**Comment by sbwalker** [comment]
@Sharafudeen you will need to fork the framework and modify every admin component as well as the control panel. This is not a feature we expect to add to the core framework.

**Comment by W6HBR** [comment]
I question why you would want to do this as the built-in security already handles limiting access.

**Reply by Sharafudeen** [reply]
Let’s say we have 5 super admins, 10 admins and 100s of moderators in one Oqtane installation with multiple applications. We wanted to temporarily block the UI across the applications. This will prevent if any wrong form entries or malfunctioning in the data level. We can alert (1 minute) all users before stopping their work.

**Reply by W6HBR** [reply]
How about implementing this through the theme?  In your default.aspx, you can have a conditional that suppresses all panes and then displays a custom message.  You could leave the Admin pane at the top for the alert message and then the subsequent notice that the site is down.

Technically, the back end would still allow activity through the API's, but I'm guessing that is probably not the concern here.

**Reply by W6HBR** [reply, marked as answer]
@Sharafudeen I just did a quick test with this and it works really well.  By not outputting the panes, it suppresses all module loads without generating any errors.  You can leave the navigation in place if needed or include that in the conditional display as well.

I might look at including something like this in my theme as well for a maintenance toggle.

**Reply by sbwalker** [reply]
@W6HBR this approach could also be used in the theme containers so that the page navigation remains active but the content in each container is disabled from entry.

**Reply by Sharafudeen** [reply]
Thanks.. FYI, I’ve seen this feature in MenaHR system.
